tests/core/test_proxy.py

```diff
@@ -44,3 +65,63 @@ def test_no_proxy(self):
         env.pop("https_proxy", None)
         env.pop("HTTP_PROXY", None)
         env.pop("HTTPS_PROXY", None)
+
+
+class CustomAgentTransaction(AgentTransaction):
+
+    def on_response(self, response):
+        super(CustomAgentTransaction, self).on_response(response)
+        self._test.stop()
```

Comment:
Nice, elegant approach. Can you set `self._test` in a constructor, in case the class is reused or refactored to somewhere else in the future?